### PR TITLE
feat: toggle component integration

### PR DESCRIPTION
**Pull request by theJohnnyMe**
- adding `tds-toggle` component
- adding listener to the custom event tdsToggle
- adding tdsToggle to types as Custom event
- adding app-toggle component to the web-component page

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,4 +1,10 @@
...
-    content: string;
-    type: 'error' | 'success' | 'information' | 'warning';
-}
\ No newline at end of file
+export type Notification = {
+  content: string;
+  type: 'error' | 'success' | 'information' | 'warning';
+};
+
+declare global {
```
[the remark this answers is not in the document]
Comment (gokberknur) [reply src/types.ts:6]: > @Tommy-Hallgren, @gokberknur, @mJarsater - I guess we should add all CustomEvents that tegel components emit? I have a lot of underlined events in WebStorom with a warning like this: `Event tdsChange is not emitted by any applicable directives nor by tds-radio-button element`


Not sure, maybe it is an IDE issue? 

Angular can not know which events host element going to listen, it is just dispatching an event. and if there is a event listener in host compoment then action will happen.

File: src/app/components/toggle/toggle.component.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  `,
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
+})
+export class ToggleComponent implements AfterViewInit {
+  @ViewChild('toggleComponent', { static: true }) toggleComponent: ElementRef;
+
+  constructor(private renderer: Renderer2) {}
+
+  ngAfterViewInit() {
+    const webComponent = this.toggleComponent.nativeElement;
```
Comment (gokberknur) [inline src/app/components/toggle/toggle.component.ts:32]: are there any reason why we select the element attach event listener here? 

for example if we remove:
 -#toggleComponent selector  
   - @ViewChild('toggleComponent', { static: true }) toggleComponent: ElementRef;
   - renderer2 injection
   
   
  and just use (tdsToggle)="listenme()" In the template. this should be enough?

File: src/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tommy-Hallgren) [reply src/types.ts:6]: i think it's an IDE issue

File: src/app/components/toggle/toggle.component.ts (hunk 2 above)
Comment (Tommy-Hallgren) [reply src/app/components/toggle/toggle.component.ts:32]: I agree, normally you do event binding directly in the template and you bind it to a function in the component.ts file.

Comment (theJohnnyMe) [reply src/app/components/toggle/toggle.component.ts]: Yep, you are right. I was looking into different options, but binding it directly in the template seems the best and most simple. I think our issue is to properly pass event types in Angular, one way is to list them down in types.ds

Comment (Tommy-Hallgren) [reply src/app/components/toggle/toggle.component.ts]: yes, you can check it out here: https://angular.io/guide/event-binding

**Review by nathalielindqvist — APPROVED** [review]
LGTM

**Comment by gokberknur** [comment]
Really great work people! 

I'm amazed how many things you implemented on demo page in such short time!

**Review by gokberknur — APPROVED** [review]

**Comment by theJohnnyMe** [comment]
> Really great work people!
> 
> I'm amazed how many things you implemented on demo page in such short time!

Thank you and Tommy for all the help and support along the journey as well as your contribution!

**Pull request merged**